diff --git a/third_party/abseil-cpp/absl/base/internal/sysinfo.cc b/third_party/abseil-cpp/absl/base/internal/sysinfo.cc
index da499d3a..2c1ef810 100644
--- a/third_party/abseil-cpp/absl/base/internal/sysinfo.cc
+++ b/third_party/abseil-cpp/absl/base/internal/sysinfo.cc
@@ -189,7 +189,11 @@ static double GetNominalCPUFrequency() {
 // and the memory location pointed to by value is set to the value read.
 static bool ReadLongFromFile(const char *file, long *value) {
   bool ret = false;
+#ifdef __MVS__
+  int fd = open(file, O_RDONLY);
+#else
   int fd = open(file, O_RDONLY | O_CLOEXEC);
+#endif
   if (fd != -1) {
     char line[1024];
     char *err;
