diff --git a/third_party/abseil-cpp/absl/base/internal/sysinfo.cc b/third_party/abseil-cpp/absl/base/internal/sysinfo.cc
index 79eaba3e..104974cf 100644
--- a/third_party/abseil-cpp/absl/base/internal/sysinfo.cc
+++ b/third_party/abseil-cpp/absl/base/internal/sysinfo.cc
@@ -198,7 +198,7 @@ static double GetNominalCPUFrequency() {
 // and the memory location pointed to by value is set to the value read.
 static bool ReadLongFromFile(const char *file, long *value) {
   bool ret = false;
-#if defined(_POSIX_C_SOURCE)
+#if defined(_POSIX_C_SOURCE) && !defined(__MVS__)
   const int file_mode = (O_RDONLY | O_CLOEXEC);
 #else
   const int file_mode = O_RDONLY;
@@ -461,6 +461,12 @@ pid_t GetTID() {
   return reinterpret_cast<pid_t>(thread);
 }
 
+#elif defined(__MVS__)
+
+pid_t GetTID() {
+  return reinterpret_cast<pid_t>((int)pthread_self().__);
+}
+
 #else
 
 // Fallback implementation of `GetTID` using `pthread_self`.
